src/code/mempak.c

```diff
@@ -1,184 +1,266 @@
+    if (osPfsInitPak(serialEventQueue, &sMempakPfsHandle, controllerNum) == 0) {
         ret = true;
     }
 
     osPfsFreeBlocks(&sMempakPfsHandle, &sMempakFreeBytes);
-    PadMgr_ReleaseSerialEventQueue(&gPadMgr, mq);
-
+    PadMgr_ReleaseSerialEventQueue(&gPadMgr, serialEventQueue);
     return ret;
 }
 
-s32 Mempak_GetFreeBytes(s32 controllerNb) {
+s32 Mempak_GetFreeBytes(s32 controllerNum) {
     return sMempakFreeBytes;
 }
 
-s32 Mempak_FindFile(s32 controllerNb, char start, char end) {
-    OSMesgQueue* mq;
+/**
+ * Checks if the files identified by letters between `start` and `end` (inclusive) exist on the memory pak.
+ *
+ * @param controllerNum Unused, the controller used is that which was last passed to `Mempak_Init`
+ * @param start Start file letter
+ * @param end End file letter (inclusive)
+ * @return a bitfield where set bits indicate that the file exists
+ */
+s32 Mempak_FindFiles(s32 controllerNum, char start, char end) {
+    OSMesgQueue* serialEventQueue;
     s32 error;
-    char idx;
+    char letter;
     u32 bit = 1;
-    s32 flag = 0;
+    s32 bits = 0;
 
-    mq = PadMgr_AcquireSerialEventQueue(&gPadMgr);
+    serialEventQueue = PadMgr_AcquireSerialEventQueue(&gPadMgr);
 
-    for (idx = start; idx <= end; idx++) {
-        sMempakExtName[0] = idx - 0x27;
+    for (letter = start; letter <= end; letter++) {
+        sMempakExtName[0] = NCH(letter);
 
         error = osPfsFindFile(&sMempakPfsHandle, sMempakCompanyCode, sMempakGameCode, sMempakGameName, sMempakExtName,
-                              &sMempakFiles[idx - 'A']);
+                              &sMempakFiles[MEMPAK_LETTER_TO_INDEX(letter)]);
         if (error == 0) {
-            flag |= bit;
+            bits |= bit;
         } else {
-            sMempakFiles[idx - 'A'] = -1;
+            sMempakFiles[MEMPAK_LETTER_TO_INDEX(letter)] = -1;
         }
 
         bit <<= 1;
-        osSyncPrintf("mempak: find '%c' (%d)\n", idx, error);
+        osSyncPrintf("mempak: find '%c' (%d)\n", letter, error);
     }
 
-    PadMgr_ReleaseSerialEventQueue(&gPadMgr, mq);
-    osSyncPrintf("mempak: find '%c' - '%c' %02x\n", start, end, flag);
-
-    return flag;
+    PadMgr_ReleaseSerialEventQueue(&gPadMgr, serialEventQueue);
+    osSyncPrintf("mempak: find '%c' - '%c' %02x\n", start, end, bits);
+    return bits;
 }
 
-s32 Mempak_Write(s32 controllerNb, char idx, void* buffer, s32 offset, s32 size) {
-    OSMesgQueue* mq;
+/**
+ * Writes data to the file identified with `letter`.
+ *
+ * @param controllerNum Unused, the controller used is that which was last passed to `Mempak_Init`
+ * @param letter Memory pak file letter, in the range 'A' to ('A' + MEMPAK_MAX_FILES)
+ * @param buffer Buffer containing data to write
+ * @param offset Offset into the file to write to
+ * @param size Size in bytes
+ * @return true if the operation completed successfully, false otherwise
+ */
+s32 Mempak_Write(s32 controllerNum, char letter, void* buffer, s32 offset, s32 size) {
```

Comment:
it's probably worth documenting that this function (as well as basically all the other functions I guess) rely on `Mempak_FindFiles` having been called, to initialize `sMempakFiles`

may be too much copypaste, may be better to put on `Mempak_FindFiles` maybe? idk